stl/inc/generator

```diff
@@ -110,16 +110,18 @@ namespace _Gen_detail {
         }
 
         template <class _Alloc2, class... _Args>
-            requires convertible_to<const _Alloc2&, _Proto_allocator>
         _NODISCARD_RAW_PTR_ALLOC static void* operator new(
             const size_t _Size, allocator_arg_t, const _Alloc2& _Al, const _Args&...) {
+            static_assert(convertible_to<const _Alloc2&, _Proto_allocator>,
+                "argument convertible to the allocator type is required");

```

Comment:
I think `"Allocator argument must be convertible to the allocator type (NXXXX [coro.generator.promise]/18)"` is more in line with our standard style for `static_assert` messages. We'll need to either block the PR until there's a new working draft or merge without the citation with a followup when there is a new working draft. I don't think there are a lot of confused users out there desperately awaiting this change, so maybe we should wait. (Also on line 124.)

Reply:
WG21-N5001 is now present. I've updated the error messages to refer it.